### PR TITLE
doc: add code documentation 

### PR DESCRIPTION
**Pull request by NagaTulasi**
ref: #14

### CODE REVIEW

File: keeper/abci.go [hunk 1]
```diff
@@ -73,6 +90,8 @@ func (h *ProofOfBlobProposalHandler) ProcessProposal(_ sdk.Context, req *abci.Re
 	return &abci.ResponseProcessProposal{Status: abci.ResponseProcessProposal_ACCEPT}, nil
 }
```
Comment (Teja2045) [inline keeper/abci.go:92]: vote extensions are also handled here. can we mention that?

File: keeper/abci.go [hunk 2]
```diff
@@ -143,6 +164,15 @@ func (k *Keeper) IsValidBlockToPostTODA(height uint64) bool {
...
 
+// aggregateVotes aggregates the stake-weighted votes for a specific range of blocks
+// from the vote extensions provided in the `ExtendedCommitInfo`. It processes the
+// votes of validators and collects the total voting power for the range.
+// Iterates through the votes in the `ExtendedCommitInfo` and processes only those votes
+// where the validator voted for the block (`BlockIDFlagCommit`).
+// For each vote, it decodes the `VoteExtension`, which contains the vote information for
+// specific block ranges.
+// If the vote extension contains a vote for the pending range, it sums the voting power
+// of validators.
```
Comment (Teja2045) [inline keeper/abci.go:175]: I think these 3 lines are enough

File: keeper/abciTypes.go [hunk 3]
```diff
@@ -2,7 +2,14 @@ package keeper
 
 import abci "github.com/cometbft/cometbft/abci/types"
 
+// StakeWeightedVotes stores the stake-weighted votes from validators and the commit info
+// for a consensus round.
 type StakeWeightedVotes struct {
-	Votes              map[string]int64
+	// A map where the key is the validator's address , and the value is the
```
Comment (Teja2045) [inline keeper/abciTypes.go:8]: key is the range of pending blocks
eg: key = "1 10"

File: keeper/abci.go [hunk 4]
```diff
@@ -73,6 +90,8 @@ func (h *ProofOfBlobProposalHandler) ProcessProposal(_ sdk.Context, req *abci.Re
 	return &abci.ResponseProcessProposal{Status: abci.ResponseProcessProposal_ACCEPT}, nil
 }
 
+// PreBlocker runs before finalizing each block, responsible for handling vote extensions
+// and managing the posting of blocks to the Avail light client.
```
Comment (Teja2045) [inline keeper/abci.go:94]: vote extensions are also handled here. Can we mention that?

**Review by Teja2045 — DISMISSED** [review]

File: chainclient/client.go [hunk 5]
```diff
@@ -4,7 +4,8 @@ const (
 	KeyringBackendTest = "test"
 )
 
-// ChainClient is client to interact with SPN.
+// ChainClient represents a client used for interacting with a network.
+// It contains configurations and credentials necessary for broadcasting transactions and interacting with the chain.
```
Comment (PrathyushaLakkireddy) [inline chainclient/client.go:8]:
```suggestion
// It contains configuration necessary for broadcasting transactions and interacting with the chain.
```

**Review by Teja2045 — APPROVED** [review]

File: keeper/abciTypes.go [hunk 6]
```diff
@@ -2,7 +2,14 @@ package keeper
 
 import abci "github.com/cometbft/cometbft/abci/types"
 
+// StakeWeightedVotes stores the stake-weighted votes from validators and the commit info
+// for a consensus round.
```
Comment (PrathyushaLakkireddy) [inline keeper/abciTypes.go:6]:
```suggestion
// StakeWeightedVotes represents the aggregated stake-weighted votes from validators,
// along with the associated commit information for a specific consensus round.
```

**Review by PrathyushaLakkireddy — APPROVED** [review]

**Pull request merged**